go/ct/common/immutable_hash_array_test.go

```diff
@@ -135,3 +136,36 @@ func TestImmutableHashArray_Get(t *testing.T) {
 		})
 	}
 }
+
+func TestINmmutableHashArray_NewRandomImmutableHashArray(t *testing.T) {
+	rnd := rand.New()
+
+	for i := 0; i < 100; i++ {
```

Comment:
Same here as for the random hash: 
 - repetition is a bit excessive
 - an all-zero value is not really a problem
 - the test would not detect if the function would always return the same non-zero value

Idea: generate N samples and check that there are no duplicates.